src/aux-server/aux-web/aux-player/scene/PlayerGame.ts

```diff
@@ -1139,23 +1139,77 @@ export class PlayerGame extends Game {
             sim.helper.transaction(asyncResult(e.taskId, null));
         }
     }
-
+    //MY CHANGES START HERE
     private _calculateViewportCoordinatesFromPosition(
         sim: Simulation,
         e: CalculateViewportCoordinatesFromPositionAction
     ) {
-        const portalTag = getPortalTag(e.portal);
+        let portalTag;
+        if (e.portal === 'map') {
+            portalTag = 'mapPortal';
+        } else if (e.portal === 'grid') {
+            portalTag = 'gridPortal';
+        } else if (e.portal === 'miniGrid') {
+            portalTag = 'miniGridPortal';
+        } else if (e.portal === 'miniMap') {
+            portalTag = 'miniMapPortal';
+        } else {
+            portalTag = getPortalTag(e.portal);
+        }
+
         const _3dSim = this._findSimulationForPortalTag(sim, portalTag);
+
         if (_3dSim) {
             const rig = _3dSim.getMainCameraRig();
             const gridScale = _3dSim.getDefaultGridScale();
-            const vector = new Vector3(
-                e.position.x * gridScale,
-                e.position.y * gridScale,
-                e.position.z * gridScale
-            );
+            const isMapPortal =
+                portalTag === 'mapPortal' || portalTag === 'miniMapPortal';
+
+            const position = {
+                x: e.position.x,
+                y: e.position.y,
+                z: e.position.z,
+            };
+
+            let vector;
+
+            if (isMapPortal) {
```

Comment:
@TroyceGowdy Instead of checking for the map portal directly, please check for the presence of a coordinate transform instead.